recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,187 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class SociConan(ConanFile):
+    name = "soci"
+    settings = "os", "arch", "compiler", "build_type"
+    description = "SOCI - The C++ Database Access Library"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://soci.sourceforge.net/"
+    license = "BSL-1.0"
+    generators = "cmake", "cmake_find_package"
+    topics = ("conan", "soci", "database")
+    exports_sources = "CMakeLists.txt", "patches/**"
+
+    short_paths = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _cmake = None
+
+    _backend_default_option = {
+        "empty": True,
+        "mysql": True,
+        "odbc": True,
+        "postgresql": True,
+        "sqlite3": True,
+        "db2": False,
+        "firebird": False,
+        "oracle": False,
+    }
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_boost": [True, False],
+    }
+    options.update({"with_backend_{}".format(backend): [True, False] for backend in _backend_default_option.keys()})
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_boost": False,
+    }
+    default_options.update({"with_backend_{}".format(backend): default for backend, default in _backend_default_option.items()})
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_backend_sqlite3:
+            self.requires("sqlite3/3.31.1")
+
+        if self.options.with_backend_mysql:
+            self.requires("libmysqlclient/8.0.17")
+
+        if self.options.with_backend_odbc:
+            self.requires("odbc/2.3.7")
+            if self.options["odbc"].with_libiconv:
+                self.requires("libiconv/1.15")
```

Comment:
`libiconv` is already a requirement of `odbc` when that option is set, no need to add it here.

Reply:
Done